File: weekThree/python/caesarCipher.py
## Ancient messages were encrypted in a very simple way - by assigning each letter a value and "rotating" that letter by a given number - this is called a "Caesar Cipher" 

## a = 1, b = 2, c = 3 etc
## after you apply a rotation of 3
## "abc" becomes "def", "cat" becomes "fdw"
## Write a function that can encrypt a string using a given number as a Caesar Cipher

# Some tips
# ord('a') -> 97
# ord('c') -> 99 == ord('a') + 2
# chr(97) -> 'a'
# ord(chr(x)) == x

#.               97 (a) <--------> 122 (z)
# 0 (?) <-----> 25 (?)
#         offset (?). <-------> offset + 25 (?) % 26
# 0 (?) <-----> 25 (?)
#.               97 (a) <--------> 122 (z)

# All inputs will be lower case letters

def CaesarCipher(input_str, offset):
    
  newStr = ''
  
  for i in range(len(input_str)):
    currChar = input_str[i]
    
    currNum = ord(currChar) - 97
    newNum = (currNum + offset) % 26
    newChar = chr(newNum + 97)
    newStr += newChar
    
  return newStr

print(CaesarCipher('acf', 4)) # egj
print(CaesarCipher('zcf', 25)) # ???

File: weekThree/java/caesarCipher.java

public class caesarCipher {
	/**
	 * Ancient messages were encrypted in a very simple way - 
	 * by assigning each letter a value and "rotating" that letter by a 
	 * given number - this is called a "Caesar Cipher" 
	 * 
	 * a = 1, b = 2, c = 3 etc
	 * after you apply a rotation of 3
	 * "abc" becomes "def", "cat" becomes "fdw"
	 */
	
	/**
	 * This method will encrypt a string using a given number as a 
	 * Caesar Cipher
	 * 
	 * all inputs will be lower case
	 * 
	 * @param String str
	 * @param int offset
	 * @return new encrypted string 
	 */
	public static String caesar_Cipher(String str, int offset) {
		String newStr = "";
		
		for(int i = 0; i < str.length(); i++) {
			char curr = str.charAt(i);
			
			int num = curr-97;
			
			int newNum = (num + offset)%26;
			
			char newChar = (char) (newNum+97);
			
			newStr+=newChar;
			
		}
		return newStr;
	}

	public static void main(String[] args) {
		System.out.println(caesar_Cipher("acf",4)); //egj
		System.out.println(caesar_Cipher("zcf",25)); //ybe
		

	}

}
